logstash-core/lib/logstash/plugin.rb

```diff
@@ -31,9 +49,21 @@ class LogStash::Plugin
   # }
   # ```
   #
-  # If you don't explicitely set this variable Logstash will generate a unique name.
+  # If you don't explicitly set this variable Logstash will generate a unique name.
   config :id, :validate => :string
 
+  # Add a set of specific operation timing thresholds for the plugin, they are used in the context
+  # of reporting slow operations.
+  #
+  # ```
+  #  thresholds => {
+  #    filter.grok.runtime => 12
+  #  }
+  # ```
+  #
+  # The values defined here takes precedence over the ones defined in the logstash config file. (default => {})
+  config :thresholds, :validate => :hash, :default => {}
```

Comment:
Where would a user learn what threshold names are available?

I also generally want us to avoid `:validate => :hash` if we can, for two reasons: First, that automated documentation has no way of helping users understand what is valid inside the hash. Second, that validation of the hash value is pretty complex and can be error prone -- the result is confused users. We see this user confusion with the `http_poller` input plugin, for example.